test/reference/reference_test.go

```diff
@@ -0,0 +1,58 @@
+package reference
+
+import (
+	"bytes"
+	"os"
+	"path"
+	"testing"
+
+	"gen/github.com/woven-planet/go-zserio/test/go/reference_modules/testobject1/testobject"
+	"github.com/icza/bitio"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func testWorkspace(filePath string) string {
+	return path.Join(os.Getenv("TEST_SRCDIR"), os.Getenv("TEST_WORKSPACE"), filePath)
+}
+
+var (
+	// ReferenceFilePath is the path to the input data.
+	ReferenceFilePath string = testWorkspace(os.Getenv("TESTDATA_BIN"))
+)
+
+func TestReencoding(t *testing.T) {
+	// Given
+	want, err := os.ReadFile(ReferenceFilePath)
+	require.NoError(t, err)
+	var (
+		got bytes.Buffer
+		testObject testobject.TestObject
+	)
+	r := bitio.NewCountReader(bytes.NewReader(want))
+
+	// When
+	require.NoError(t, testObject.UnmarshalZserio(r), "unmarshal zserio")
+	w := bitio.NewCountWriter(&got)
+	require.NoError(t, err, testObject.MarshalZserio(w), "marshal zserio")
+	require.NoError(t, w.Close(), "flush buffer")
+
+	// Then
+	assert.Equal(t, want, got.Bytes())
+}
+
+func TestEqual(t *testing.T) {
```

Comment:
The test names are a bit confusing for me - both tests test for equality. Perhaps rename to `TestRoundtripFromZserio` and `TestRoundtripFromInstance` ?